lib/pdf/reader/standard_security_handler.rb

```diff
@@ -87,9 +89,19 @@ def decrypt( buf, ref )
       objKey = @encrypt_key.dup
       (0..2).each { |e| objKey << (ref.id >> e*8 & 0xFF ) }
       (0..1).each { |e| objKey << (ref.gen >> e*8 & 0xFF ) }
-      length = objKey.length < 16 ? objKey.length : 16
-      rc4 = RC4.new( Digest::MD5.digest(objKey)[0,length] )
-      rc4.decrypt(buf)
+      if @cfm == :AESV2
+        objKey << 'sAlT'  # Algorithm 1, b)
+        length = objKey.length < 16 ? objKey.length : 16
+        cipher = OpenSSL::Cipher.new("AES-#{length << 3}-CBC")
+        cipher.decrypt
+        cipher.key = Digest::MD5.digest(objKey)[0,length]
+        cipher.iv = buf[0..15]
+        cipher.update(buf[16..-1]) + cipher.final
+      else
+        length = objKey.length < 16 ? objKey.length : 16
+        rc4 = RC4.new( Digest::MD5.digest(objKey)[0,length] )
+        rc4.decrypt(buf)
+      end
     end
```

Comment:
Regarding more extensive refactoring, there is another issue to consider.

PDF 1.5 spec allows that streams and strings may use different crypt filters. It's very rare but still legal. So I was thinking that either `decrypt` takes a flag to distinguish stream vs string or `build_security_handler` creates two filters - one for stream and one for string (even though most of the time, both points to the same filter)

So, for now, I will break `StandardSecurityHandler#decrypt` into smaller functions